packages/devtools_app/test/flutter/logging_screen_test.dart

```diff
@@ -0,0 +1,81 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:devtools_app/src/globals.dart';
+import 'package:devtools_app/src/logging/flutter/logging_screen.dart';
+import 'package:devtools_app/src/logging/logging_controller.dart';
+import 'package:devtools_app/src/service_manager.dart';
+import 'package:devtools_app/src/ui/flutter/service_extension_widgets.dart';
+import 'package:devtools_app/src/utils.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+
+import '../support/mocks.dart';
+import 'wrappers.dart';
+
+void main() {
+  LoggingScreen screen;
+  group('Info Screen', () {
+    MockLoggingController mockLoggingController;
+    Widget wrap(Widget widget) =>
+        wrapWithControllers(widget, loggingController: mockLoggingController);
+
+    setUp(() async {
+      await ensureInspectorDependencies();
+      mockLoggingController = MockLoggingController();
+      when(mockLoggingController.data).thenReturn([]);
+      when(mockLoggingController.onLogsUpdated).thenReturn(Notifier());
+
+      setGlobal(
+        ServiceConnectionManager,
+        FakeServiceManager(useFakeService: true),
+      );
+      when(serviceManager.connectedApp.isDartWebApp)
+          .thenAnswer((_) => Future.value(false));
+
+      screen = const LoggingScreen();
+    });
+
+    testWidgets('builds its tab', (WidgetTester tester) async {
+      await tester.pumpWidget(wrap(Builder(builder: screen.buildTab)));
+      expect(find.text('Logging'), findsOneWidget);
+    });
+
+    testWidgets('builds with no data', (WidgetTester tester) async {
+      await tester.pumpWidget(wrap(Builder(builder: screen.build)));
+      expect(find.byType(LoggingScreenBody), findsOneWidget);
+      expect(find.text('Clear logs'), findsOneWidget);
+      expect(find.byType(StructuredErrorsToggle), findsOneWidget);
+    });
+
+    testWidgets('can clear logs', (WidgetTester tester) async {
+      await tester.pumpWidget(wrap(Builder(builder: screen.build)));
+      verifyNever(mockLoggingController.clear());
+      await tester.tap(find.text('Clear logs'));
+      verify(mockLoggingController.clear()).called(1);
+    });
+
+    testWidgets('can toggle structured errors', (WidgetTester tester) async {
+      await tester.pumpWidget(wrap(Builder(builder: screen.build)));
+    });
```

Comment:
add logic to toggle structured errors on/off

Reply:
Good catch; done.